pom.xml

```diff
@@ -65,6 +65,7 @@ under the License.
         <netty.version>4.1.100.Final</netty.version>
         <jackson.version>2.15.3</jackson.version>
         <guava.version>32.1.3-jre</guava.version>
+        <flink.markBundledAsOptional>true</flink.markBundledAsOptional>
```

Comment:
doesn't it make sense to upgrade the shade plugin to `3.4.1` here as well instead of having it only being upgraded for the jackson dependencies? 

my understanding is (according to [your comment in FLINK-34148](https://issues.apache.org/jira/browse/FLINK-34148?focusedCommentId=17809796&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17809796)) that the shade plugin stopped editing the dependency tree in 3.4.1. 

Maven 3.3+ didn't allow editing the dependency tree anymore (based on the description in FLINK-28016). Shouldn't we have run into issues already earlier because we upgraded Maven in CI to 3.8.2 even though we were still using an older version of the shade plugin? It seems that I'm missing something here. :thinking:

Reply:
If I understand things around [MNG-5899](https://issues.apache.org/jira/browse/MNG-5899) correctly in maven they stopped reuse dependency reduced pom produced by shade plugin however they didn't forbid manipulation on api level.
At the same time on api level it is still possible to get some info e.g. list of dependencies from maven and then make some manipulation in code with this list of dependencies, this is how it was in maven-shade-plugin [1]. After [MSHADE-413](https://issues.apache.org/jira/browse/MSHADE-413) instead of manipulation on collection of objects received from maven they started to clone it first and then perform same manipulation on cloned version

[1] https://github.com/apache/maven-shade-plugin/pull/124/files#diff-083ef87367e79a2ea6b333846149588f4ce284ed930db1d87712fdbf9111e381L1066-L1112

Reply:
I see, those were different issue. But that still leaves the question open whether we want to enable the more recent shade plugin version for all modules. :thinking:

Reply:
that was the plan 
the only thing stopped me from making it here is that I was thinking about making ShadeOptionalChecker working which then could be backported to 18.x and then as a separate task sync maven-shade-plugin for all the modules

Or do you think it worth making it here?

Reply:
fair point :+1: a follow-up Jira task is fine as well.

Reply:
yep, sure FLINK-34269